app/questionnaire/path_finder.py

```diff
@@ -37,51 +37,73 @@ def is_path_complete(self, path):
 
     def get_first_incomplete_location(self, path):
         if path:
-            first_location = path[0]
-            section_id = self.schema.get_section_id_for_block_id(
-                first_location.block_id
-            )
-
             for location in path:
                 block = self.schema.get_block(location.block_id)
                 block_type = block.get('type')
-                if location not in self.progress_store.get_completed_locations(
-                    section_id
-                ) and block_type not in ['SectionSummary', 'Summary', 'Confirmation']:
+
+                is_location_complete = (
+                    location
+                    in self.progress_store.get_completed_locations(
+                        section_id=location.section_id,
+                        list_item_id=location.list_item_id,
+                    )
+                )
+
+                if not is_location_complete and block_type not in [
+                    'SectionSummary',
+                    'Summary',
+                    'Confirmation',
+                ]:
                     return location
 
         return None
 
-    def full_routing_path(self):
+    def full_routing_path(self, sections=None):
         path = []
-        sections = self.schema.get_sections()
-        for section in sections:
-            path = path + list(self.routing_path(section))
+        schema_sections = sections or self.schema.get_sections()
+
+        for section_schema in schema_sections:
+            section_id = section_schema['id']
+            for_list = self.schema.get_repeating_list_for_section(section_id)
```

Comment:
Nothing major again, just a slight dislike of the name 'for_list', not sure its represents what it is

Reply:
Yeah, not the easiest, 'section_repeating_list' is a mouthful, but the best I have at the minute.

Reply:
Ah the problem with that is that for_list is used elsewhere for other actions, such as in relationship collectors

Reply:
So I used the name `for_list` since that is what we call it in the schema.
I've gone with `repeating_list` in the end.